credsweeper/app.py

```diff
@@ -299,6 +300,16 @@ def data_scan(self, data_provider: DataContentProvider, depth: int, recursive_li
             except Exception as zip_exc:
                 # too many exception types might be produced with broken zip
                 logger.error(f"{data_provider.file_path}:{zip_exc}")
+
+        elif Util.is_gzip(data_provider.data):
+            try:
+                with gzip.open(io.BytesIO(data_provider.data)) as f:
+                    gzip_content_provider = DataContentProvider(data=f.read(), file_path=data_provider.file_path)
```

Comment:
No check for excluded files?

Reply:
it will be throw through exception. 
excluding extension is removed from list when --depth enabled

Reply:
Regarding excluded files there is `check_exclude_file()` method in zip file case.

https://github.com/Samsung/CredSweeper/blob/70b15c72e3cb8c613a1d2065505706cf7148c4bb/credsweeper/app.py#L288-L289

But not in gzip case..

https://github.com/Samsung/CredSweeper/blob/70b15c72e3cb8c613a1d2065505706cf7148c4bb/credsweeper/app.py#L304-L311

I think it can be solve with `FileForceExtractor`(temporary name..), that I wrote below.

Reply:
There is other issue - if --depth is enabled, does the user want to scan obviously archives (zip,gz,...) or not? Probably only APK, or XLSX ...
As you can see --depth reads file anyway to recognize the type. Extension may be various or absent..

Reply:
data_provider.file_path - yes, there is no workaround to remove *.gz if exists.